template/faas-flow/handler.go

```diff
@@ -817,50 +817,36 @@ func handleDynamicEnd(fhandler *flowHandler, context *faasflow.Context, result [
 	}
+	// Get unique execution id of the node
+	branchkey := pipeline.GetNodeExecutionUniqueId(currentNode) + "-branch-completion"
 
 	// if indegree is > 1 then use statestore to get indegree completion state
 	if len(options) > 1 {
 
 		// Get unique execution id of the node
 		key = pipeline.GetNodeExecutionUniqueId(currentNode) + "-branch-completion"
 		// Update the state of indegree completion and get the updated state
-		lastIndegree, err := fhandler.RetriveCounter(key)
-		if err != nil {
-			return []byte(""), fmt.Errorf("failed to RetriveCounter inDegree counter for node %s", currentNode.GetUniqueId())
-		}
-
-		// not all the branches finished executing
-		// most situation context.Set will run len(option) times
-		// but we don't conext.Get last one
-		if inDegreeUpdatedCount+1 < len(options) {
 
-			// Skip if dynamic node data forwarding is not disabled
-			if currentNode.GetForwarder("dynamic") != nil {
-				option := pipeline.CurrentDynamicOption[currentNode.GetUniqueId()]
-				key = fmt.Sprintf("%s--%s--%s", option, pipeline.GetNodeExecutionUniqueId(currentNode), currentNode.GetUniqueId())
+		// Skip if dynamic node data forwarding is not disabled
+		if currentNode.GetForwarder("dynamic") != nil {
+			option := pipeline.CurrentDynamicOption[currentNode.GetUniqueId()]
+			key = fmt.Sprintf("%s--%s--%s", option, pipeline.GetNodeExecutionUniqueId(currentNode), currentNode.GetUniqueId())
 
-				err = context.Set(key, result)
-				if err != nil {
-					return nil, fmt.Errorf("failed to store branch result of dynamic node %s for option %s, error %v",
-						currentNode.GetUniqueId(), option, err)
-				}
-				fmt.Printf("[Request `%s`] Intermidiate result from Branch to Dynamic Node %s for option %s stored as %s\n",
-					fhandler.id, currentNode.GetUniqueId(), option, key)
+			err = context.Set(key, result)
+			if err != nil {
+				return nil, fmt.Errorf("failed to store branch result of dynamic node %s for option %s, error %v",
+					currentNode.GetUniqueId(), option, err)
 			}
-
-			fmt.Printf("[Request `%s`] Executing end of dynamic node %s, delayed as "+
-				"completed indegree: %d/%d\n",
-				fhandler.id, currentNode.GetUniqueId(), inDegreeUpdatedCount, len(options))
-			// Return nil response making current iteration to stop
-			return nil, nil
+			fmt.Printf("[Request `%s`] Intermidiate result from Branch to Dynamic Node %s for option %s stored as %s\n",
+				fhandler.id, currentNode.GetUniqueId(), option, key)
 		}
-		realIndegree, err = fhandler.IncrementCounter(key, 1)
+		realIndegree, err := fhandler.IncrementCounter(branchkey, 1)
 		if err != nil {
 			return []byte(""), fmt.Errorf("failed to update inDegree counter for node %s", currentNode.GetUniqueId())
 		}
 
 		fmt.Printf("[Request `%s`] Executing end of dynamic node %s, completed indegree: %d/%d\n",
-			fhandler.id, currentNode.GetUniqueId(), inDegreeUpdatedCount, len(options))
+			fhandler.id, currentNode.GetUniqueId(), realIndegree, len(options))
```

Comment:
You also need delete the key for current branch
as we skipping it inside for loop
```go
// skip retrieving data for current option
if option == currentOption {
          continue
}
```
It would not be deleted
Just adding a `context.Del(key)` after the indegree check would work